robotd/master.py

```diff
@@ -14,155 +14,170 @@
 from robotd.devices import BOARDS
 
 
-def _send(connection, message):
-    # Split the message into buf_size chunks
-    buf_size = BoardRunner.SOCK_BUFFER_SIZE
-    for msg in [message[i:i + buf_size] for i in range(0, len(message), buf_size)]:
-        connection.send(msg)
+class Connection:
+
+    def __init__(self, socket):
+        self.socket = socket
+        self.file = socket.makefile('r', encoding='utf-8')
+
+    def close(self):
+        self.socket.close()
```

Comment:
Worth implementing the context manager protocol in the spirit of least surprise?